src/Sentry.Maui.CommunityToolkit.Mvvm/MauiCommunityToolkitMvvmEventsBinder.cs

```diff
@@ -0,0 +1,149 @@
+
+            case RefreshView refresh:
+                TryBindTo(refresh.Command, bind);
+                break;
+
+            case SearchBar searchBar:
+                TryBindTo(searchBar.SearchCommand, bind);
+                break;
+
+            default:
+                TryGestureBinding(element, bind);
+                break;
+        }
+    }
+
+    private void TryGestureBinding(VisualElement element, bool bind)
+    {
+        if (element is IGestureRecognizers gestureRecognizers)
+        {
+            foreach (var gestureRecognizer in gestureRecognizers.GestureRecognizers)
+            {
+                TryBindTo(gestureRecognizer, bind);
+            }
+        }
+    }
+
+    private void TryBindTo(IGestureRecognizer recognizer, bool bind)
+    {
+        switch (recognizer)
+        {
+            case TapGestureRecognizer tap:
+                TryBindTo(tap.Command, bind);
+                break;
+
+            case SwipeGestureRecognizer swipe:
+                TryBindTo(swipe.Command, bind);
+                break;
+
+            case DragGestureRecognizer drag:
+                TryBindTo(drag.DragStartingCommand, bind); // unlikely to ever be async
+                TryBindTo(drag.DropCompletedCommand, bind);
+                break;
+
+            case PointerGestureRecognizer pointer:
+                TryBindTo(pointer.PointerPressedCommand, bind);
+                TryBindTo(pointer.PointerReleasedCommand, bind);
+                TryBindTo(pointer.PointerEnteredCommand, bind); // unlikely to ever be async
+                TryBindTo(pointer.PointerExitedCommand, bind);
+                break;
+
+                // no command bindings on these gestures, so they're left out
+                // PinchGestureRecognizer
+                // PanGestureRecognizer
+        }
+    }
+
+    private void TryBindTo(ICommand? command, bool bind)
+    {
+        if (command is IAsyncRelayCommand relayCommand)
+        {
+            // since events can retrigger binding pickups, we want to ensure we unhook any previous event handlers
+            // instead of storing a ref to know if we've already bound to an event or not
+            relayCommand.PropertyChanged -= RelayCommandOnPropertyChanged;
+            if (bind)
+            {
+                relayCommand.PropertyChanged += RelayCommandOnPropertyChanged;
+            }
+        }
+    }
+
+
+    private readonly ConcurrentDictionary<IAsyncRelayCommand, ISpan> _contexts = new();
+    private const string SpanName = "ctmvvm";
+    private const string SpanOp = "relay.command";
+
+    private void RelayCommandOnPropertyChanged(object? sender, PropertyChangedEventArgs e)
+    {
+        if (e.PropertyName == nameof(IAsyncRelayCommand.IsRunning))
+        {
+            var relay = (IAsyncRelayCommand)sender!;
+
+            if (relay.IsRunning)
+            {
+                var span = hub.StartSpan(SpanName, SpanOp);
+                if (span is ITransactionTracer transaction)
```

Comment:
There's something very unusual going on here. The first time this event triggers, there's no transaction on the scope and so a transaction gets created here.

Further down on line 145 (after the transaction is finished) the scope has been correctly reset and there's no transaction on the scope anymore.

So far so good...

However, the second time this event triggers, for some reason there's a transaction on the scope... and so a span gets created here (rather than a transaction). 

The result is that this integration currently only ever creates one trace per AsyncRelayCommand! 🐞 

No running theories currently - needs investigation.

Reply:
Ah, I think I see what's going on here. Most stuff on the Scope is global in MAUI cause we IsGlobalMode is always true for MAUI apps. Transactions are the exception:
https://github.com/getsentry/sentry-dotnet/blob/8abc247566132a4b284d7fe24c44a93d84d32076/src/Sentry/Scope.cs#L203

And in this particular instance, we're creating spans/transactions in for an AsyncRelayCommand... it seems to start on the main thread but the completion happens on a different thread so on the first run through, everything looks fine (it looks like we reset the transaction on the scope). On the second run through we're in a different asynclocal context and so the transaction hasn't been reset on the scope (it's Scope.Transaction is `AsyncLocal<T>`). 

Tricky...

Reply:
I think the simplest solution is not to store transactions representing AsyncRelayCommands on the Scope... 

This means other integrations (like DB integrations or our SentryMessageHandler) won't create child spans for these.

I can't think of any other solutions off the top of my head though.